docs/storage/includes/storage-app-host.md

```diff
@@ -0,0 +1,142 @@
+
+> [!IMPORTANT]
+> When you call <xref:Aspire.Hosting.AzureStorageExtensions.AddAzureStorage*>, it implicitly calls <xref:Aspire.Hosting.AzureProvisionerExtensions.AddAzureProvisioning*>—which adds support for generating Azure resources dynamically during app startup. The app must configure the appropriate subscription and location.
+
+#### Generated provisioning Bicep
+
+If you're new to [Bicep](/azure/azure-resource-manager/bicep/overview), it's a domain-specific language for defining Azure resources. With .NET Aspire, you don't need to write Bicep by-hand, instead the provisioning APIs generate Bicep for you. When you publish your app, the generated Bicep is output alongside the manifest file. When you add an Azure Storage resource, the following Bicep is generated:
+
+<!-- markdownlint-disable MD033 -->
+<br/>
+<details>
+<summary id="storage-bicep"><strong>Toggle Azure Storage bicep.</strong></summary>
+<p aria-labelledby="storage-json">
+
+:::code language="bicep" source="../../snippets/azure/AppHost/storage.module.bicep":::
+
+</p>
+</details>
+<!-- markdownlint-enable MD033 -->
+
+The preceding Bicep is a module that provisions an Azure Storage account with the following defaults:
+
+- `kind`: The kind of storage account. The default is `StorageV2`.
+- `sku`: The SKU of the storage account. The default is `Standard_GRS`.
+- `properties`: The properties of the storage account:
+  - `accessTier`: The access tier of the storage account. The default is `Hot`.
+  - `allowSharedKeyAccess`: A boolean value that indicates whether the storage account permits requests to be authorized with the account access key. The default is `false`.
+  - `minimumTlsVersion`: The minimum supported TLS version for the storage account. The default is `TLS1_2`.
+  - `networkAcls`: The network ACLs for the storage account. The default is `{ defaultAction: 'Allow' }`.
+
+In addition to the storage account, it also provisions a blob container. The following role assignments are also added, where all roles are scoped to the storage account—and the roles are [built-in Azure role-based access control (Azure RBAC) roles](/azure/role-based-access-control/built-in-roles#storage):
+
+- Storage Blob Data Contributor `(ba92f5b4-2d11-453d-a403-e96b0029c9fe)`: Read, write, and delete Azure Storage containers and blobs.
+- Storage Table Data Contributor `(0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3)`: Read, write, and delete Azure Storage tables and entities.
+- Storage Queue Data Contributor `(974c5e8b-45b9-4653-ba55-5f855dd0fb88)`: Read, write, and delete Azure Storage queues and queue messages.
+
+The generated Bicep is a starting point and can be customized to meet your specific requirements. For more information on provisioning, see [Local Azure provisioning](../../deployment/azure/local-provisioning.md).
+
+### Add Azure Storage emulator resource
```

Comment:
Generally explaining that in the RunAsEmulator callback, you can configure the azurite container resource would be a good thing to add. People want to change the image tag or azurite or set the ports or make the container persistent. We should show how to do that.